openedx/openedx-platform
# Paver Removal 3/3

## Description

This fully removes Paver from edx-platform and updates some documentation accordingly. See individual commits for more detail.

* BREAKING CHANGE: Removes all remaining Paver commands including
`pavelib/prereqs.py:*` and `pavelib/assets.py:*`.

* BREAKING CHANGE: Removes `./manage.py [lms|cms] compile_sass`, which
was just a wrapper around Paver commands.

* BREAKING CHANGE: Removes paver.txt. Operators should install testing.txt
instead.

* BREAKING CHANGE: Removes libsass from requirements/edx/base.txt.
Operators will need to install requirements/edx/assets.txt in order to
compile Sass.

Note that the `build!: rejigger npm run test commands` commit is technically a breaking change, but [those commands were first added a week ago](https://github.com/openedx/edx-platform/pull/35159) and were not documented, so I don't feel like we need to raise this as BREAKING CHANGE.

## Supporting information

This plan has been accepted and communicated via the DEPR process:
* https://github.com/openedx/edx-platform/issues/34467

Previous PRs (blocking this one):
* https://github.com/openedx/edx-platform/pull/34829
* ~https://github.com/openedx/edx-platform/pull/34830~
* https://github.com/openedx/edx-platform/issues/34845 
* https://github.com/overhangio/tutor/pull/1178
* https://github.com/openedx/openedx-tutor-plugins/pull/31
* https://github.com/openedx/edx-platform/pull/36046

## Testing instructions

I built the Tutor openedx and openedx-dev images with this branch. I smoke tested LMS and CMS in both modes, and ensure that I could toggle between Indigo and the default theme.

I inspected the JS Test logs to ensure that we're still running 6 test suites (jest, cms-vanilla, cms-require, xmodule-vanilla, common-vanilla, common-require).

Finally, you can check out the updated edx-platform testing guide here: https://github.com/kdmccormick/edx-platform/blob/kdmccormick/paver-rm-3/docs/concepts/testing/testing.rst. Using Tutor, I installed the tutor-contrib-legacy-js plugin (from a branch: `https://github.com/kdmccormick/openedx-tutor-plugins/tree/kdmccormick/test-legacy-js/plugins/tutor-contrib-test-legacy-js`) and ran JS tests locally.
